gcc-6.3 + glibc-2.25 now can detect 2-argument open that is supposed to created new files.
diff --git a/tests/api/fsio.c b/tests/api/fsio.c
index 2041f43..18d173b 100644
--- a/tests/api/fsio.c
+++ b/tests/api/fsio.c
@@ -1058,3 +1058,3 @@ START_TEST (fsio_sys_access_file_test) {
   /* Make the file to check; we want it to have perms 664.*/
-  fd = open(fsio_test_path, O_CREAT|O_EXCL|O_WRONLY);
+  fd = open(fsio_test_path, O_CREAT|O_EXCL|O_WRONLY, S_IRUSR | S_IWUSR);
   fail_if(fd < 0, "Unable to create file '%s': %s", fsio_test_path,
diff --git a/tests/api/scoreboard.c b/tests/api/scoreboard.c
index f0ffdbc..f26d833 100644
--- a/tests/api/scoreboard.c
+++ b/tests/api/scoreboard.c
@@ -267,3 +267,3 @@ START_TEST (scoreboard_lock_test) {
 
-  fd = open(test_file2, O_CREAT|O_EXCL|O_RDWR);
+  fd = open(test_file2, O_CREAT|O_EXCL|O_RDWR, S_IRUSR | S_IWUSR);
   fail_unless(fd >= 0, "Failed to open '%s': %s", test_file2, strerror(errno));
@@ -911,3 +911,3 @@ START_TEST (scoreboard_entry_lock_test) {
 
-  fd = open(test_file2, O_CREAT|O_EXCL|O_RDWR);
+  fd = open(test_file2, O_CREAT|O_EXCL|O_RDWR, S_IRUSR | S_IWUSR);
   fail_unless(fd >= 0, "Failed to open '%s': %s", test_file2, strerror(errno));
